reccmp/isledecomp/dir.py

```diff
@@ -12,6 +13,53 @@ def winepath_unix_to_win(path: str) -> str:
     return subprocess.check_output(["winepath", "-w", path], text=True).strip()
 
 
+def _iter_path_components(path: PurePath) -> Iterator[str]:
+    """Walk path components in reverse and convert to lower case for matching."""
+    for p in reversed(path.parts):
+        yield p.lower()
+
+
+def _count_matching_path_parts(
+    remote_path: PurePath, local_path: PurePath
+) -> tuple[int, PurePath]:
+    score = 0
+    for rp, lp in zip(
+        _iter_path_components(remote_path), _iter_path_components(local_path)
+    ):
+        # Don't try to resolve any dot directories.
+        # We would get it wrong if any of the paths are symlinks.
+        if rp != lp or rp in (".", "..") or lp in (".", ".."):
+            break
+
+        score += 1
+
+    return (score, local_path)
+
+
+def convert_foreign_path(
+    remote_path: PurePath, local_paths: tuple[PurePath]
+) -> PurePath | None:
+    """Connect the given remote_path to the best match from the list of local_paths.
+    For best performance, you should narrow down the starting list to paths
+    that match the base filename."""
+    scored = [_count_matching_path_parts(remote_path, p) for p in local_paths]
+    scored.sort(reverse=True)
+
+    if len(scored) >= 2:
+        [(top_score, top_path), (next_score, _)] = scored[:2]
+        # Return if this is the best match above all others
+        if top_score > next_score:
+            return top_path
```

Comment:
Minor nitpick: One could add something like
```py
else:
    # The two best matches have the same score. We return None because we can't know which one is correct
    return None
```
to make the behaviour a bit more explicit